libraries/go/webhook.go

```diff
@@ -77,23 +78,23 @@ func (wh *Webhook) verify(payload []byte, headers http.Header, enforceTolerance
 	msgSignature := headers.Get(HeaderWebhookSignature)
 	msgTimestamp := headers.Get(HeaderWebhookTimestamp)
 	if msgId == "" || msgSignature == "" || msgTimestamp == "" {
-		return errRequiredHeaders
+		return fmt.Errorf("unable to verify payload, err: %w", ErrRequiredHeaders)
 	}
 
 	timestamp, err := parseTimestampHeader(msgTimestamp)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to verify payload, err: %w", err)
 	}
 
 	if enforceTolerance {
 		if err := verifyTimestamp(timestamp); err != nil {
-			return err
+			return fmt.Errorf("unable to verify payload, err: %w", err)
```

Comment:
why don't we return `ErrRequiredHeaders` ? It would allow the consumer to do error comparison